File: Snapshot_GO/app/snapshotbitcake/ly_marker_message.go
package snapshotbitcake

import (
	"Snapshot/app"
	"Snapshot/servent/message"
	"strconv"
)

type LYMarkerMessage struct {
	// This weird way of passing a basic message is for class extending(this class extends BasicMessage).
	message.BasicMessage
}

func ConstructLYMarkerMessage(
	sender *app.ServentInfo,
	receiver *app.ServentInfo,
	collectorId int,
) LYMarkerMessage {
	return LYMarkerMessage{
		// Pass to the "super" constructor.
		*message.ConstructBasicMessage("LY_MARKER", sender, receiver, strconv.Itoa(collectorId)),
	}
}

File: src/servent/message/snapshot/LYMarkerMessage.java
package servent.message.snapshot;

import app.ServentInfo;
import servent.message.BasicMessage;
import servent.message.MessageType;

public class LYMarkerMessage extends BasicMessage {

	private static final long serialVersionUID = 388942509576636228L;

	private Integer snapshotNumber;

	private LYTellMessage lyTellMessage;

	private Integer collectorId;

	public LYMarkerMessage(int snapshotNumber, ServentInfo sender, ServentInfo receiver, int collectorId, LYTellMessage lyTellMessage) {
		super(MessageType.LY_MARKER, sender, receiver, String.valueOf(collectorId));

		this.snapshotNumber = snapshotNumber;
		this.lyTellMessage = lyTellMessage;
		this.collectorId = collectorId;
	}

	public Integer getSnapshotNumber() {
		return snapshotNumber;
	}

	public LYTellMessage getLyTellMessage() {
		return lyTellMessage;
	}

	public Integer getCollectorId() {
		return collectorId;
	}
}
